pkg/webhook/vlanconfig/validator.go

```diff
@@ -143,6 +145,21 @@ func (v *Validator) Delete(_ *admission.Request, oldObj runtime.Object) error {
 		return fmt.Errorf(deleteErr, vc.Name, err)
 	}
 
+	nads, err := v.nadCache.List(StorageNetworkNetAttachDefNamespace, labels.Set(map[string]string{
+		utils.KeyClusterNetworkLabel: vc.Spec.ClusterNetwork,
+	}).AsSelector())
+	if err != nil {
+		return fmt.Errorf(deleteErr, vc.Name, err)
+	}
+
+	if len(nads) > 0 {
+		for _, nad := range nads {
+			if nad.Annotations[StorageNetworkAnnotation] == "true" {
```

Comment:
to be more secure,  please skip the case where `nad.DeletionTimestamp != nil`

Reply:
sure added the changes